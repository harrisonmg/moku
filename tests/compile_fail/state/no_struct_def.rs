use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod machine {}

    struct Top;
    impl TopState<BlinkyState> for Top {}

    struct Bottom;

    type BottomTy = Bottom;

    #[superstate(Top)]
    impl State<BlinkyState> for BottomTy {}
}

fn main() {}
