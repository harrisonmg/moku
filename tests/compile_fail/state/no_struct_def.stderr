error: could not find the struct definition for this State in this module
  --> tests/compile_fail/state/no_struct_def.rs:18:33
   |
18 |     impl State<BlinkyState> for BottomTy {}
   |                                 ^^^^^^^^
